src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1440,14 +1442,31 @@ protected void AfterMembersChecks(DiagnosticBag diagnostics)
                 }
 
                 // https://github.com/dotnet/roslyn/issues/30080: Report diagnostics for base type and interfaces at more specific locations.
-                var baseType = BaseTypeNoUseSiteDiagnostics;
-                var interfaces = InterfacesNoUseSiteDiagnostics();
                 if (baseType?.NeedsNullableAttribute() == true ||
-                    interfaces.Any(t => t.NeedsNullableAttribute()))
+                     interfaces.Any(t => t.NeedsNullableAttribute()))
                 {
                     compilation.EnsureNullableAttributeExists(diagnostics, location, modifyCompilation: true);
                 }
             }
+
+            if (needsTupleElementNamesAttribute(baseType) ||
```

Comment:
>if (needsTupleElementNamesAttribute(baseType) || [](start = 12, length = 48)

I didn't see changes like that made on VB side. Were we already reporting errors for scenarios like this for VB?